### ISSUE TITLE
Questions about evaluation of Shap-E

### ISSUE THREAD

**Issue by YueWuHKUST** [crockwell/Cap3D#11]
I followed the instructions on https://github.com/crockwell/Cap3D/tree/main/text-to-3D/shap-e to generate multi-view images. 

However, the result turns out to be in black background. 
![33e39520b6204975a177156ab6c4a25c_0](https://github.com/crockwell/Cap3D/assets/17807074/7b6af2d0-699b-47d4-8e09-47bb13c8903d)

While the provided ground truth image is in white background.
![0a5a52e4339545e9b34744353863efd2_0](https://github.com/crockwell/Cap3D/assets/17807074/69110fb5-5357-4a7b-95fb-6631faa799c5)

Am I missing anything to make the test result correct? This will influence the calculation of CLIP score and 2D FID.

**Comment by tiangeluo** [comment]
Did you download the shap-E code from our repository or from the original shap-E repository?

There are modifications deeply inside the shap-E code of our repository to make the background grey.

**Comment by YueWuHKUST** [comment]
I followed your advice, and downloaded your latest code to run the rendering of multi-views images. But the results still turns out to be in black background.

After downloading your code, I only made a simple modifications in ./text-to-3D/shap-e/shap_e/util/collections.py
Change from 
```
class AttrDict(OrderedDict):
```
to:
```
K = TypeVar('K')
V = TypeVar('V')

class AttrDict(OrderedDict[K, V], Generic[K, V]):
```
Otherwise, the code will throw an error.

After conducting this simple modification, I run the test but the results is still in black background.

![33e39520b6204975a177156ab6c4a25c_0](https://github.com/crockwell/Cap3D/assets/17807074/81f86c81-6a6e-40b6-b423-0692dc6dcae9)

**Comment by tiangeluo** [comment]
Hi @YueWuHKUST , I came up with a better rendering strategy to avoid the modification from the shap-E codes, which is first generating `.ply` files and then render it (the same as PointE). 

By incorporating this modification, I will need to update the numbers. I am working on this plan, and hopefully get back to you tonight or tomorrow.

**Comment by YueWuHKUST** [comment]
Yes, I'm running with ```render_type stf```. And the warning message says, since there is no pytorch3d, it will use the pytorch renderer. I thought the difference between pytorch3d render and pytorch renderer is pytorch3d supports full gradients? Their performance should be similar?

Let me try to install pytorch3d and update you with the result with stf rendering.

I tried with ```--render_type=nerf```, and the result is grey background, but is slightly different from the groudtruth image style, that these images contains black lines around object, while the ground truth does not have.
![7cd963a9d0e845f3a7a6fa4c843c7e21_0](https://github.com/crockwell/Cap3D/assets/17807074/e70ef3e7-5442-4924-b7b3-a6c2f77f3f51)

BTW, your modifications inside the code should only influence the inferece process, right? Is it ok for me to finetune the pretrained model with the original code?

**Comment by tiangeluo** [comment]
Hi @YueWuHKUST ,

> Let me try to install pytorch3d and update you with the result with stf rendering.

Thanks. The black background according to my understanding is due to the missing of pytorch3d. I look forward to your updates.

> the result is grey background, but is slightly different from the groudtruth image style, that these images contains black lines around object, while the ground truth does not have.

We're also considering this point. I'm thinking of adding a comparison using meshes rendered via Blender, as done in PointE. This procedure will make the rendered images sharing the exact same background color as the ground-truth and also no black contour.

 I've implemented the functions and plan to provide one more line of comparisons in Table 6. Will inform you once I have done this.
- https://github.com/crockwell/Cap3D/blob/main/text-to-3D/shap-e/text2ply_shapE.py
- https://github.com/crockwell/Cap3D/blob/main/text-to-3D/shap-e/render_script_shapE.py

> BTW, your modifications inside the code should only influence the inferece process, right? Is it ok for me to finetune the pretrained model with the original code?

My modification will not affect finetune part. It is the same as the original code.

**Comment by YueWuHKUST** [comment]
I tried to use pytorch3d for the rendering. The result is grey background and has no black contour. But from my understanding, the ground truth images are rendered using blender on mesh, so the test images generated by Shap-E will be more reasonable if use blender to render the multi-view images.

Another question is, is your modification inside **point-E** will influence the rendering result of Point-E ? The rendering is done by blender, thus I use the original code of Point-E to render the result, and the result seems ok.

**Comment by tiangeluo** [comment]
Sounds good for the pytorch3d. 

Shap-E currently saved mesh via 'stf' mode; therefore, using blender to render the saved mesh is another way to calculate the numbers for 'stf' mode performance. I'll add one more line of numbers for the stf-mesh-blender way. However, note that, we currently have no practical way to compute similar scores to 'nerf' mode.

What do you mean "use the original code of Point-E to render the result"? My current way is first saving `.ply` and then using Blender to render. I did not use any of the code from Point-E to render.

**Comment by YueWuHKUST** [comment]
Sounds good. 

I mean since inside your Shap-E's code, you did some modifications to make the background grey. So I want to know what modifications have you made inside Point-E's code? Will this modifications influence the inference result？

**Comment by tiangeluo** [comment]
I see. No worries, my modification will not interfere the Point-E inference.

**Comment by YueWuHKUST** [comment]
Sounds good !
